src/App.tsx

```diff
@@ -1,29 +1,47 @@
 import './App.css';
-
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import TodoList from './Components/ToDoList';
-import { ITodo } from './Shared/types';
-import { data } from './Shared/sample.json';
+import { IToDoResp } from './Shared/types';
+import useFetch from './Components/useFetch';
 
 function App() {
-  const toDoArray:ITodo[] = data.sampleDataForList;
+  const { data, loading, error } = useFetch('https://jsonplaceholder.typicode.com/todos');
+
+  useEffect(() => {
+    if (data) {
+      setTodos(data);
+    }
+  }, [data]);
+
+
+  const toDoArray:IToDoResp[] | null = data;
+  
   const [todos, setTodos] = useState(toDoArray);
```

Comment:
remove this you are already setting API value in state inside useEffect

Reply:
Resolved